examples/ble_passive_scanning/advertisement.cpp

```diff
@@ -1,16 +1,18 @@
 #include "advertisement.h"
 
-Advertisement::Advertisement() : advertisementType_(0), len_(0) {
+Advertisement::Advertisement() {
+  std::memset(&header_, 0, HEADER_SIZE);
   std::memset(&address_, 0, ADDRESS_SIZE);
-  std::memset(&data_, 0, DATA_SIZE);
+  std::memset(&data_, 0, DATA_MAX_SIZE);
 }
 
-Advertisement::Advertisement(const unsigned char* buf, int len) : advertisementType_(buf[ADV_TYPE]), len_(len)
+Advertisement::Advertisement(const unsigned char* buf, int len)
 {
-  std::memcpy(&address_, &buf[2], ADDRESS_SIZE);
-  if (len_ > DATA_SIZE) {
-    std::memcpy(&data_, &buf[8], len_ - DATA_SIZE);
-  }
+  std::memcpy(&header_, &buf[HEADER_START], HEADER_SIZE);
+  std::memcpy(&address_, &buf[ADDRESS_START], ADDRESS_SIZE);
+  unsigned char data_len = len - HEADER_SIZE - ADDRESS_SIZE;
+  if (data_len > DATA_MAX_SIZE) data_len = DATA_MAX_SIZE;
```

Comment:
style grumble: I actually prefer braces but I'm no "real" C++ developer

Reply:
Strongly agree. Non-brace if statements lead to future bugs.